test/malli/roundtrip_test.clj

```diff
@@ -17,43 +17,49 @@
     (is (nil? (rt/explain-roundtrip [:keyword])))
     (is (nil? (rt/explain-roundtrip [:nil])))
     (is (nil? (rt/explain-roundtrip [:any])))
-    (is (nil? (rt/explain-roundtrip [:number])))
-    (is (nil? (rt/explain-roundtrip [:enum "a" "b" "c"]))))
+    (is (nil? (rt/explain-roundtrip number?)))
+    (is (nil? (rt/explain-roundtrip int?)))
+    (is (nil? (rt/explain-roundtrip [:enum "a" "b" "c"])))))
 
 (deftest roundtrippable-or-nonoverlapping
   (testing ":or with non-overlapping branches is roundtrippable"
     (is (nil? (rt/explain-roundtrip [:or [:int] [:string]])))
     (is (nil? (rt/explain-roundtrip [:or [:keyword] [:string]])))))
 
 (deftest roundtrippable-or-overlapping
-  (testing ":or with overlapping branches is not roundtrippable"
-    (let [result (rt/explain-roundtrip [:or [:int] [:number]])]
-      (is (vector? result))
-      (is (seq result))
-      (is (some #(re-find #"overlap" (:problem %)) result)))
-    (let [result (rt/explain-roundtrip [:or [:number] [:int]])]
+  (testing ":or with overlapping simple-parser branches IS roundtrippable"
+    ;; When both branches have simple (non-transforming) parsers,
+    ;; there's no ambiguity even if types overlap
+    (is (nil? (rt/explain-roundtrip [:or [:int] number?])))
+    (is (nil? (rt/explain-roundtrip [:or number? [:int]]))))
+  (testing ":or with overlapping non-simple branches is NOT roundtrippable"
+    ;; When branches have transforming parsers and overlap, it's not roundtrippable
+    (let [result (rt/explain-roundtrip [:or [:orn [:i [:int]]] [:orn [:j [:int]]]])]
       (is (vector? result))
       (is (seq result))
       (is (some #(re-find #"overlap" (:problem %)) result)))))
```

Comment:
Please verify all return values of `explain-roundtrip` with either `nil?` or `=` to replicate and document the exact error message.

e.g.,
```clojure
(is (= [...] (rt/explain-roundtrip [:or [:orn [:i [:int]]] [:orn [:j [:int]]]])))
```

Don't bother with an explicit `vector?` check, just know that's part of the API of `explain-roundtrip` to return a nilable vector.